src/base64.h

```diff
@@ -129,12 +129,6 @@ static size_t base64_encode(const char* src,
                             size_t slen,
                             char* dst,
                             size_t dlen) {
-  // We know how much we'll write, just make sure that there's space.
-  CHECK(dlen >= base64_encoded_size(slen) &&
-        "not enough space provided for base64 encode");
-
-  dlen = base64_encoded_size(slen);
```

Comment:
The CHECK is to ensure that the caller upholds its part of the contract: that the destination buffer is large enough.

It's allowed to call it with a destination buffer that is much larger than needed, that's why the actual number of bytes written is returned.

Hope that clears it up.